src/System.Net.NetworkInformation/tests/FunctionalTests/IPInterfacePropertiesTest_Linux.cs

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Net.Sockets;
+using System.Net.Test.Common;
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.Net.NetworkInformation.Tests
+{
+    [PlatformSpecific(PlatformID.Linux)]
+    public class IPInterfacePropertiesTest_Linux
+    {
+        private readonly ITestOutputHelper _log;
+
+        public IPInterfacePropertiesTest_Linux()
+        {
+            _log = TestLogging.GetInstance();
+        }
+
+        [Fact]
+        public void IPInfoTest_AccessAllProperties_NoErrors()
+        {
+            foreach (NetworkInterface nic in NetworkInterface.GetAllNetworkInterfaces())
+            {
+                _log.WriteLine("Nic: " + nic.Name);
+                _log.WriteLine("- Supports IPv4: " + nic.Supports(NetworkInterfaceComponent.IPv4));
+                _log.WriteLine("- Supports IPv6: " + nic.Supports(NetworkInterfaceComponent.IPv6));
+
+                IPInterfaceProperties ipProperties = nic.GetIPProperties();
+
+                Assert.NotNull(ipProperties);
+
+                Assert.Throws<PlatformNotSupportedException>(() => ipProperties.AnycastAddresses);
+
+                Assert.NotNull(ipProperties.DhcpServerAddresses);
+                _log.WriteLine("- Dhcp Server Addresses: " + ipProperties.DhcpServerAddresses.Count);
+                foreach (IPAddress dhcp in ipProperties.DhcpServerAddresses)
+                {
+                    _log.WriteLine("-- " + dhcp.ToString());
```

Comment:
> The primary value of these tests seems to be that we're exercising the code and verifying that exceptions aren't thrown where there not supposed to be, right?

That's correct. There's no simple way to implement these tests given that information retrieved is machine dependent.

> and the logging can probably be enabled, or implemented in another way (just make a simple ITestOutputHelper for Unix)

I thought that `EventSource` is already supported in *NIX. In this case, XUnit's default implementation of `ITestOutputHelper` might actually work since it's not invoked on a callback (thread-pool thread), outside the context of a test. If that ever happens, XUnit's default implementation will crash the entire test process (the main reason I've used EventSource).